model/shared/src/main/scala/explore/model/Asterism.scala

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.model
+
+import cats.Eq
+import cats.data.NonEmptyList
+import cats.syntax.all._
+import lucuma.core.model.Target
+import monocle._
+
+final case class Asterism(targets: NonEmptyList[TargetWithId]) {
+  def toSidereal: List[SiderealTargetWithId] =
+    targets.traverse(_.toSidereal).foldMap(_.toList)
+
+  // This should be calculatedd from the other targets or manually overriden
+  def baseTarget: TargetWithId = targets.head
+
+  def hasId(id: Target.Id): Boolean = targets.exists(_.id === id)
+}
+
+object Asterism {
+  val isoTargets: Iso[NonEmptyList[TargetWithId], Asterism] =
+    Iso[Asterism, NonEmptyList[TargetWithId]](_.targets)(Asterism.apply).reverse
+
+  val targets: Lens[Asterism, NonEmptyList[TargetWithId]] = isoTargets.reverse.asLens
+
+  val fromTargets: Lens[NonEmptyList[TargetWithId], Asterism] = isoTargets.asLens
```

Comment:
I think we can omit these two lenses, they can be created at usage site from the Iso, if need be.

Reply:
👍